.github/workflows/ruby.yml

```diff
@@ -21,7 +21,7 @@ on:
   push:
     paths:
       - '.github/workflows/ruby.yml'
-      - 'ci/docker/**'
+      - 'ci/docker/linux-apt-ruby.dockerfile'
```

Comment:
We would need to list all dockerfiles used in the parent images, in this case:

```
ci/docker/ubuntu-18.04-cpp.dockerfile
ci/docker/linux-apt-c-glib.dockerfile
ci/docker/linux-apt-ruby.dockerfile
```

though if we wan't to run `UBUNTU=20.04 archery docker run ubuntu-ruby` then:

```
ci/docker/ubuntu-20.04-cpp.dockerfile
ci/docker/linux-apt-c-glib.dockerfile
ci/docker/linux-apt-ruby.dockerfile
```

This hierarchy is encoded in the [docker-compose.yml](https://github.com/apache/arrow/blob/master/docker-compose.yml#L69) depending on the env [variables as well](https://github.com/apache/arrow/blob/master/docker-compose.yml#L542).